packages/hash/frontend/src/pages/[account-slug]/types/entity-type/property-list-card.tsx

```diff
@@ -76,70 +93,104 @@ export const PropertyTypeRow = ({
-            background: theme.palette.gray[10],
-          },
-        }),
-      ]}
-    >
-      <TableCell>
-        <Typography variant="smallTextLabels" fontWeight={500}>
-          {property.title}
-        </Typography>
-      </TableCell>
-      <TableCell>
-        <PropertyExpectedValues property={property} />
-      </TableCell>
+          }),
+        ]}
+      >
+        <TableCell>
+          <Typography variant="smallTextLabels" fontWeight={500}>
+            {property.title}
+          </Typography>
+        </TableCell>
+        <TableCell>
+          <PropertyExpectedValues property={property} />
+        </TableCell>
 
-      <MultipleValuesCell propertyIndex={propertyIndex} />
+        <MultipleValuesCell propertyIndex={propertyIndex} />
 
-      <CenteredTableCell sx={{ textAlign: "center" }}>
-        <Controller
-          render={({ field: { value, ...field } }) => (
-            <Checkbox {...field} checked={value} />
-          )}
-          control={control}
-          name={`properties.${propertyIndex}.required`}
-        />
-      </CenteredTableCell>
+        <CenteredTableCell sx={{ textAlign: "center" }}>
+          <Controller
+            render={({ field: { value, ...field } }) => (
+              <Checkbox {...field} checked={value} />
+            )}
+            control={control}
+            name={`properties.${propertyIndex}.required`}
+          />
+        </CenteredTableCell>
 
-      <TableCell
-        sx={{
-          [`.${iconButtonClasses.root}`]: {
-            opacity: 0,
-            [`.${tableRowClasses.root}:hover &`]: {
-              opacity: 1,
+        <TableCell
+          sx={{
+            [`.${iconButtonClasses.root}`]: {
+              opacity: 0,
+              [`.${tableRowClasses.root}:hover &`]: {
+                opacity: 1,
+              },
             },
-          },
+          }}
+        >
+          <PropertyMenu
+            editButtonProps={bindTrigger(editModalPopupState)}
+            onRemove={onRemove}
+            property={property}
+            popupState={menuPopupState}
+          />
+        </TableCell>
+      </TableRow>
+      <PropertyTypeForm
+        popupState={editModalPopupState}
+        modalTitle={<>Edit Property Type</>}
+        onSubmit={async (data) => {
+          // @todo verify this works
+          await updatePropertyType({
+            data: {
+              propertyTypeId: $id,
+              propertyType: formDataToPropertyType(data),
+            },
+          });
+
+          await refetchPropertyTypes?.();
+
+          onUpdatePropertyTypeVersion($id);
         }}
-      >
-        <PropertyMenu
-          onRemove={onRemove}
-          property={property}
-          popupState={menuPopupState}
-        />
-      </TableCell>
-    </TableRow>
+        submitButtonProps={{ children: <>Edit property type</> }}
+        fieldProps={{ name: { disabled: true } }}
+        getDefaultValues={() => ({
```

Comment:
This is a function in order to make the crash for any unhandled property types happen when you try to click edit, instead of at render time. This was negotiated with @luisbettencourt in order to make combining this PR with his property arrays PR easier.